ykcompile/src/lib.rs

```diff
@@ -266,7 +278,7 @@ impl<TT> TraceCompiler<TT> {
                                 );
                             }
                             self.free_if_temp(curloc);
-                            curloc = Location::Register(temp, store);
+                            curloc = Self::reg_or_addr(temp, store);
```

Comment:
Since this is always preceded by `if !store`, should we remove `reg_or_addr` and do something like:
```
if !store {
    dynasm!(self.asm
     ; mov Rq(temp), [Rq(temp)]
    );
    curloc = Location::Register(temp);
 } else {
  curloc = Location::Address(temp);
}
```
I don't know if that makes things clearer.